classes.fields.php

```diff
@@ -213,6 +222,47 @@ public function boolean_attr( $attrs = array() ) {
 	}
 
 	/**
+	 * Check for validation array key in CMB Field.
+	 *
+	 * @return boolean
+	 */
+	public function should_validate_values() {
+		return ( ! empty( $this->args['validation'] ) );
+	}
+
+	/**
+	 * Return escaped parsley attributes
+	 * supports: required, required => minlength, required => maxlength
+	 *
+	 * @param array $attrs
+	 *
+	 * @todo Support all Parsley default validation types.
+	 */
+	public function validation_attr( $attrs = array() ) {
+
+		if ( ! $this->should_validate_values() ) {
+			return;
+		}
+
+		$validation_rules = wp_parse_args( $attrs, $this->args['validation'] );
+
+		// Validation rule: Required.
+		if ( isset( $validation_rules['required'] ) ) {
+
+			echo esc_attr( ' required ' );
+
+			if ( is_array( $validation_rules['required'] ) ) {
```

Comment:
Thanks @mattheu. I agree with the changes and will do a separate PR to add parsley via bower (so the file count isn't awful on this PR).

Visual styling: I peeked at ACF and their implementation of visual styling is quite nice, so I might grab a bit of that as well.

Agree with the `config` and `attr` changes as well - thats perfect.

Hoping to add support for custom validation messages via the validation array as well so it becomes something like:

```
'validation => [
    'required' => [
        'message' => __esc_html( 'You will need to input something here..'),
    ]
]
```

Live notifications seems like a win as well. So I'll add that :)

Will keep moving on the implementation when time allows, but happy to keep you in the loop along the way.

Truly appreciate the feedback here :heart: